src/AlgebraicPetri.jl

```diff
@@ -272,6 +272,21 @@ inputs(p::AbstractPetriNet, t) = subpart(p, incident(p, t, :it), :is)
 """
 outputs(p::AbstractPetriNet, t) = subpart(p, incident(p, t, :ot), :os)
 
+""" induced_subnet(p::T, t)
+
+Return the induced subnet consisting of the transition(s) in `t`, all input
+places and input arcs, and all output places and output arcs. The returned
+object will be of the same type as `p`.
+"""
+function induced_subnet(p::T, t) where {T <: AbstractPetriNet}
+  induced_subnet(p, [t])
+end
+
+function induced_subnet(p::T, t::AbstractVector{Int}) where {T <: AbstractPetriNet}
+  subnet = Subobject(p, T=t)
+  return dom(hom(~(¬subnet)))
```

Comment:
Should we return the Subobject and have the caller call dom if they want just the domain?

We could add overloads for Subobject -> PetriNet so that they can call `PetriNet(induced_subnet(g))` if they don't want the morphism?

Reply:
@jpfairbanks those all sound like good ideas, but I don't know how we'd want to implement the overload for `PetriNet` to pull the domain subobject out. `PetriNet` is a specific type made by calling the `@acset_type` macro, it's plausible a user would want to get the domain petri net out of a subobject where the petri nets in question are of a different type, like a labelled petri net. How should the overload work in that case?

Reply:
Would `T(p::Subobject{T}) where T<:AbstractPetriNet = dom(p)` work?